typing/ctype.ml

```diff
@@ -698,22 +698,23 @@ let duplicate_class_type ty =
    [expand_abbrev] (via [subst]) requires these expansions to be
    preserved. Does it worth duplicating this code ?
 *)
-let rec generalize ty =
-  let level = get_level ty in
-  if (level > !current_level) && (level <> generic_level) then begin
+let rec generalize ~level ty =
```

Comment:
In the `generalize` function the meaning of `level` might be not completely obvious from the name since we are not creating types at level `level`  (like in `newstub`, `copy` or the various `instance` functions) but generalizing types with level above this level.

I wonder if we should explicit this distinction by naming the  `generalize` argument `above_level` or something similar?